pipeline_dp/accumulator.py

```diff
@@ -0,0 +1,117 @@
+import abc
+import typing
+import pickle
+from functools import reduce
+
+
+def merge(accumulators: typing.Iterable[
+  'Accumulator'] = None) -> 'Accumulator':
+  """Merges the accumulators.
+
+  Args:
+    accumulators:
+
+  Returns: Accumulator instance with merged values.
+  """
+  unique_accumulator_types = {type(accumulator).__name__ for accumulator in
+                              accumulators}
+  if len(unique_accumulator_types) > 1:
+    raise TypeError(
+      "Accumulators should all be of the same type. Found accumulators of "
+      + f"different types: ({','.join(unique_accumulator_types)}).")
+
+  return reduce(lambda acc1, acc2: acc1.add_accumulator(acc2), accumulators)
+
+
+class Accumulator(abc.ABC):
+  """Base class for all accumulators.
+
+    Accumulators are objects that encapsulate aggregations and computations of
+    differential private metrics.
+  """
+
+  @abc.abstractmethod
+  def add_value(self, value):
+    """Adds the value to each of the accumulator.
+    Args:
+      value: value to be added.
+
+    Returns: self.
+    """
+    pass
+
+  @abc.abstractmethod
+  def add_accumulator(self, accumulator: 'Accumulator') -> 'Accumulator':
+    """Merges the accumulator to self and returns self.
```

Comment:
Please add comment:

That sub-class implementation is responsible for checking that types of self and accumulator are the same.

Reply:
Added